bulk_grades/api.py

```diff
@@ -437,6 +437,38 @@ def get_rows_to_export(self):
                     row[f'grade-{block_id}'] = effective_grade
             yield row
 
+    def filtered_column_headers(self):
+        """
+        To trim down grade exports, only show subsections which were modified in a bulk update.
+        Returns: a filtered list of columns to export, preserving modified and non-subsection columns
+        """
+        unmodified_subsections = set()
+        for subsection in self._subsections.keys():
+            unmodified_subsections.add(subsection)
+
+        for row in self.result_data:
+            # Ignore rows which didn't introduce changes
+            if row['status'] == 'No Action':
+                continue
+
+            modified_by_row = set()
+
+            # Get changes this row introduced for an as-yet-unmodified subsection
+            for subsection in unmodified_subsections:
+                if row[f'new_override-{subsection}'] != '':
+                    modified_by_row.add(subsection)
+
+            # Remove modified subsections from the unmodified list
+            if modified_by_row:
+                unmodified_subsections.difference_update(modified_by_row)
```

Comment:
for some reason i just love using these set operation methods

Reply:
I must say, they are cool.